### PR TITLE
Adds Bughunt, Incursion, Spy vs Spy, Tower Defense

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
- rscadd: "Added three new gamemodes: Tower Defense, Bughunt, Spy vs. Spy."
  - rscadd: "Bughunt, mercenary and borer, needs twenty required players and five required enemies."
  - tweak: "Borers have received some miscellaneous fixes and tweaks across the board. They should overall work better now."
  - rscadd: "Borers can now awaken psionics in their hosts. Maybe it's not a bad idea to get a brain slug after all..."
  - rscadd: "Spy vs. Spy, traitor and renegade, needs four required players."
  - tweak: "Changed how renegades work a bit. You're not traitors! You're paranoid people who smuggled a gun to work and are working to keep themselves safe."
  - rscadd: "Tower Defense, mercenary, heist and ninjas, needs thirty required players and ten required enemies. Good luck!"

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by GMR25** [comment]
Isn't 'Threesome' more than a tad inappropriate name-wise?

**Comment by NonQueueingMatt** [comment]
!review

**Comment by Lolziam** [comment]
Are borers still able to use tools such as welders to unweld doors?

**Comment by NonQueueingMatt** [comment]
> 
> 
> Are borers still able to use tools such as welders to unweld doors?

Borers aren't human mobs, so no.

File: code/modules/mob/living/simple_animal/borer/borer_powers.dm [hunk 1]
```diff
@@ -282,83 +270,215 @@
...
+	if(chemicals < 75)
+		to_chat(src, span("warning", "You don't have enough chemicals!"))
+		return
+
+	var/list/faculties = list(capitalize(PSI_COERCION), capitalize(PSI_REDACTION), capitalize(PSI_ENERGISTICS), capitalize(PSI_PSYCHOKINESIS))
+	var/selected_faculty = input(src, "Choose a faculty to upgrade.") as null|anything in faculties
+	if(!selected_faculty)
+		return
+	selected_faculty = lowertext(selected_faculty)
+	if(host.psi.get_rank(selected_faculty) >= PSI_RANK_MASTER)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/borer/borer_powers.dm]:
```suggestion
	if(host.psi.get_rank(selected_faculty) >= PSI_RANK_GRANDMASTER)
```

File: code/modules/mob/living/simple_animal/borer/borer_powers.dm [hunk 2]
```diff
@@ -282,83 +270,215 @@
...
+		to_chat(src, span("notice", "This faculty has already been pushed to the max potential you can achieve!"))
+		return
+
+	chemicals -= 75
+	to_chat(src, span("notice", "You probe your tendrils deep within your host's zona bovinae, seeking to upgrade their abilities."))
+	to_chat(host, span("warning", "You feel a burning tingling sensation at the back of your head."))
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/borer/borer_powers.dm]:
```suggestion
	to_chat(host, span("warning", "You feel a burning, tingling sensation at the back of your head..."))
```

File: code/game/antagonist/antagonist_add.dm [hunk 3]
```diff
@@ -33,7 +33,8 @@
 	if(faction_verb && player.current)
 		player.current.verbs |= faction_verb
 
-	player.current.client.verbs += /client/proc/aooc
+	if(player.current.client)
+		player.current.client.verbs += /client/proc/aooc
```
Comment (skull132) [inline code/game/antagonist/antagonist_add.dm]:
```dm
player.current?.client.verbs += /client/proc/aooc
```

succinct

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 4]
```diff
@@ -150,6 +150,15 @@ emp_act
 				return 1
 	return 0
 
+/mob/living/carbon/human/proc/check_head_airtight_coverage()
+	var/list/clothing = list(head, wear_mask, wear_suit)
+	for(var/obj/item/clothing/C in clothing)
+		if(!istype(C))
+			continue
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_defense.dm]: this is not possible in a `for(var/sometype/T in X)` loop. Nix it.

File: code/modules/mob/living/simple_animal/borer/borer_captive.dm [hunk 5]
```diff
@@ -32,20 +31,21 @@
...
-			to_chat(H, "<span class='danger'>With an immense exertion of will, you regain control of your body!</span>")
-			to_chat(B.host, "<span class='danger'>You feel control of the host brain ripped from your grasp, and retract your probosci before the wild neural impulses can damage you.</span>")
-			B.detatch()
-			verbs -= /mob/living/carbon/proc/release_control
-			verbs -= /mob/living/carbon/proc/punish_host
-			verbs -= /mob/living/carbon/proc/spawn_larvae
+	..()
 
+/mob/living/captive_brain/proc/eject_borer(var/mob/living/simple_animal/borer/borer, var/mob/living/captive_brain/host)
+	if(!borer || !borer.controlling)
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/borer/borer_captive.dm]:
```suggestion
	if(!borer?.controlling)
```

File: code/modules/mob/living/simple_animal/borer/borer_powers.dm [hunk 6]
```diff
@@ -282,83 +270,215 @@
...
+
+//Brain slug proc for voluntary removal of control.
+/mob/living/carbon/proc/release_control()
+	set category = "Abilities"
+	set name = "Release Control"
+	set desc = "Release control of your host's body."
+
+	var/mob/living/simple_animal/borer/B = has_brain_worms()
+
+	if(B && B.host_brain)
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/borer/borer_powers.dm]:
```suggestion
	if(B?.host_brain)
```

**Pull request merged**